taxonomy-editor-frontend/src/pages/editentry/AccumulateAllComponents.jsx

```diff
@@ -14,46 +15,104 @@ import ListAllEntryRelationships from "./ListAllEntryRelationships";
     }
+
+    // Loading...
     if (isPending) {
-        return (
-            <Box>
-                <Typography sx={{ml: 4}} variant='h5'>Loading..</Typography>
-            </Box>
-        )
+        return (<Typography sx={{ml: 4}} variant='h5'>Loading..</Typography>)
+    }
+    
+    // Helper functions for Dialog component 
+    const handleClose = () => {setOpen(false)};
+    const handleBack = () => {navigate('/entry')};
+
+    // Function handling updation of node
+    const handleSubmit = () => {
+        const {id, ...data} = updatedNodeObject // ID not allowed in POST
+        fetch(url, {
+            method : 'POST',
+            headers: {"Content-Type" : "application/json"},
+            body: JSON.stringify(data)
+        }).then(() => {
+            if (isEntry) {
+                fetch(url+'children/', {
+                    method : 'POST',
+                    headers: {"Content-Type" : "application/json"},
+                    body: JSON.stringify(updateChildren)
+                }).then(() => {
+                    setOpen(true);
+                }).catch((errorMessage) => {
+                    console.log(errorMessage);
+                })
+            }
+        }).catch((errorMessage) => {
+            console.log(errorMessage);
+        })
     }
     return ( 
         <Box className="node-attributes">
             {/* Based on isEntry, respective components are rendered */}
             { isEntry ? 
                 <Box>
                     { !!nodeObject &&
-                        <>  <ListAllEntryRelationships url={url+'parents'} title={'Parents'} />
-                            <ListAllEntryRelationships url={url+'children'} title={'Children'} />
-                            <ListTranslations nodeObject={nodeObject} setNodeObject={setNodeObject} /> 
-                            <ListAllEntryProperties nodeObject={nodeObject} setNodeObject={setNodeObject} /> </> }
+                        <>  <ListEntryParents url={url+'parents'} />
+                            <ListEntryChildren url={url+'children'} setUpdateNodeChildren={setUpdateChildren} />
+                            <ListAllEntryProperties nodeObject={updatedNodeObject} setNodeObject={setUpdatedNodeObject} originalNodeObject={nodeObject} /> </> }
                 </Box> :
-                <>  <ListAllNonEntryInfo nodeObject={nodeObject} id={id} setNodeObject={setNodeObject} /> </>
+                <>  <ListAllNonEntryInfo nodeObject={updatedNodeObject} id={id} setNodeObject={setUpdatedNodeObject} originalNodeObject={nodeObject} /> </>
             }
+            {/* Button for submitting edits */}
+            <Button
+                variant="contained"
+                onClick={handleSubmit}
+                sx={{ml: 4, mt:2, width: 130}}>
+                    Submit
+            </Button>
+            {/* Dialog box for acknowledgment of update */}
+            <Dialog
```

Comment:
what about using a snackbar: https://mui.com/material-ui/react-snackbar/#customization instead of the Dialog, I feel is a simpler way of notifying the user

Reply:
Since this is a priority notification, I thought it should have a dialog of its own. Having a snackbar for acknowledgement wouldn't be ideal from a user's point of view.